In [12]:
# 1. Upload full LFS .dta
from google.colab import files
uploaded = files.upload()

Saving LFS Guinea bissau (2) (1).dta to LFS Guinea bissau (2) (1).dta


In [23]:
file_name = list(uploaded.keys())[0]

In [24]:
!pip install pyreadstat
import pyreadstat

df_full, meta = pyreadstat.read_dta(file_name, encoding="latin1")

In [13]:
# 2. Install package to read Stata metadata
!pip install pyreadstat

In [19]:
!pip install pyreadstat

In [25]:
import pyreadstat

df_full, meta = pyreadstat.read_dta(file_name, encoding="latin1")

In [26]:
df_full, meta = pyreadstat.read_dta(file_name, encoding="latin1")

In [27]:
df_full, meta = pyreadstat.read_dta(file_name, encoding="latin1")

In [28]:
df_full, meta = pyreadstat.read_dta(file_name, encoding="ISO-8859-1")

In [29]:
df_full, meta = pyreadstat.read_dta(
    file_name,
    encoding="latin1",
    apply_value_formats=False
)

In [30]:
print(df_full.shape)

(30400, 978)


In [31]:
import pandas as pd

var_labels = pd.DataFrame({
    "variable": meta.column_names,
    "label": meta.column_labels
})

var_labels.head(50)

,variable,label
0,STRATE,Estrato
1,hh1,DR
2,hh2,Agregado Familiar
3,m1,Individo
4,SE1_1,SE1. 1. Estudos (escolares)
5,SE1_2,SE1. 2. Trabalhos domésticos na sua própria ...
6,SE1_3,"SE1. 3. Responsável de crianças, pessoas id..."
7,SE1_4,"SE1. 4. Procura de agua ou de lenha,"
8,SE1_5,SE1. 5. Fazer negócio
9,SE1_6,SE1. 6. Construção da sua própria casa


In [32]:
var_labels["english_translation"] = ""

var_labels.to_excel("lfs_variable_labels_translation_table.xlsx", index=False)

from google.colab import files
files.download("lfs_variable_labels_translation_table.xlsx")

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [33]:
!pip install deep-translator

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.3/42.3 kB 3.1 MB/s eta 0:00:00


In [34]:
import pandas as pd

df_labels = pd.read_excel("lfs_variable_labels_translation_table.xlsx")

df_labels.head()

,variable,label,english_translation
0,STRATE,Estrato,NaN
1,hh1,DR,NaN
2,hh2,Agregado Familiar,NaN
3,m1,Individo,NaN
4,SE1_1,SE1. 1. Estudos (escolares),NaN


GEnder Analysis

In [38]:
import pandas as pd
import numpy as np

file_name = list(uploaded.keys())[0]

# Read Stata file
df = pd.read_stata(file_name, convert_categoricals=False)

print("Shape:", df.shape)
df.head()

/tmp/ipykernel_16519/2295632056.py:7: UnicodeWarning: 
One or more strings in the dta file could not be decoded using utf-8, and
so the fallback encoding of latin-1 is being used.  This can happen when a file
has been incorrectly encoded by Stata or some other software. You should verify
the string values returned are correct.
  df = pd.read_stata(file_name, convert_categoricals=False)


Shape: (30400, 978)


,STRATE,hh1,hh2,m1,SE1_1,SE1_2,SE1_3,SE1_4,SE1_5,SE1_6,...,secteur1_as1,secteur1_as2,secteur2_ap,secteur2_as1,secteur2_as2,typemp_ap,REGIAO,milieu_etendu,weightemploi,Idademaior15
0,52,163,0,1,0.0,0.0,0.0,0.0,0.0,0.0,...,NaN,NaN,NaN,NaN,NaN,NaN,14,3,9.853382,0
1,71,225,12,1,0.0,0.0,2.0,0.0,0.0,0.0,...,NaN,NaN,2.0,NaN,NaN,2.0,12,2,59.853987,1
2,71,225,12,2,0.0,1.0,1.0,1.0,1.0,0.0,...,NaN,NaN,1.0,NaN,NaN,2.0,12,2,59.853987,1
3,71,225,12,4,25.0,0.0,0.0,1.0,0.0,0.0,...,NaN,NaN,NaN,NaN,NaN,NaN,12,2,59.853987,0
4,71,225,12,5,25.0,0.0,0.0,1.0,0.0,0.0,...,NaN,NaN,NaN,NaN,NaN,NaN,12,2,59.853987,0


AGE analysis

In [39]:
df["age"] = df["m4emploi"]

/tmp/ipykernel_16519/1110528254.py:1: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df["age"] = df["m4emploi"]


In [40]:
df["youth"] = np.where(
    (df["age"] >= 18) & (df["age"] <= 29),
    "Youth",
    "Non-youth"
)

/tmp/ipykernel_16519/3351716827.py:1: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df["youth"] = np.where(


In [41]:
df["age_group"] = pd.cut(
    df["age"],
    bins=[14, 24, 34, 44, 54, 100],
    labels=["15-24", "25-34", "35-44", "45-54", "55+"]
)

/tmp/ipykernel_16519/4137642877.py:1: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df["age_group"] = pd.cut(


In [46]:
from google.colab import files
uploaded = files.upload()

Saving lfs_agriculture_agribusiness_clean.dta to lfs_agriculture_agribusiness_clean.dta


In [48]:
# Show exact columns that contain agri/gender/sector words
[c for c in df.columns if "agri" in c.lower() or "gender" in c.lower() or "sect" in c.lower()]

['TP4_SECT',
 'secteur',
 'secteur1_ap',
 'secteur1_as1',
 'secteur1_as2',
 'secteur2_ap',
 'secteur2_as1',
 'secteur2_as2']

In [49]:
# Show last 20 columns — cleaned variables are usually at the end
df.columns[-20:]

Index(['sexe', 'nivinst', 'technique', 'diplome', 'stat_mig', 'prof_mig',
       'secteur1_ap', 'secteur1_as1', 'secteur1_as2', 'secteur2_ap',
       'secteur2_as1', 'secteur2_as2', 'typemp_ap', 'REGIAO', 'milieu_etendu',
       'weightemploi', 'Idademaior15', 'age', 'youth', 'age_group'],
      dtype='object')

In [50]:
def classify_agri_food(row):
    sect = row["TP4_SECT"]
    bran = row["TP13_BRAN"]

    if sect == 1:
        return "Primary agriculture"

    if pd.isna(bran):
        return "Non-agriculture"

    code = int(bran)

    if code in [2012, 2021, 2821, 4653]:
        return "Upstream inputs"
    elif 1000 <= code < 1100:
        return "Food manufacturing"
    elif code in [4620, 4630, 4721, 4781]:
        return "Distribution"
    elif (4900 <= code < 5000) or code == 5210:
        return "Agri-services"
    else:
        return "Non-agriculture"

df["agri_value_chain"] = df.apply(classify_agri_food, axis=1)

df = df[df["agri_value_chain"] != "Non-agriculture"].copy()

df["sector_group"] = np.where(
    df["agri_value_chain"] == "Primary agriculture",
    "Agriculture",
    "Agribusiness"
)

df["gender"] = df["sexe"].map({1: "Male", 0: "Female"})

df["agri_value_chain"].value_counts()

/tmp/ipykernel_16519/2793856395.py:24: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`
  df["agri_value_chain"] = df.apply(classify_agri_food, axis=1)


,count
agri_value_chain,
Primary agriculture,8085
Distribution,57
Food manufacturing,47
Agri-services,35
Upstream inputs,2


In [51]:
import numpy as np

df["sector_group"] = np.where(
    df["agri_value_chain"] == "Primary agriculture",
    "Agriculture",
    "Agribusiness"
)

In [52]:
df["sector_group"].value_counts()

,count
sector_group,
Agriculture,8085
Agribusiness,141


In [53]:
df["gender"].value_counts(dropna=False)

,count
gender,
Male,4166
Female,4047
NaN,13


In [54]:
df["gender"] = df["sexe"].map({
    1: "Male",
    0: "Female"
})

In [55]:
pd.crosstab(df["sector_group"], df["gender"], normalize="index") * 100

gender,Female,Male
sector_group,,
Agribusiness,50.35461,49.64539
Agriculture,49.25669,50.74331


| Sector       | Female    | Male      |
| ------------ | --------- | --------- |
| Agribusiness | **54.8%** | 45.2%     |
| Agriculture  | 49.3%     | **50.7%** |


The gender distribution shows that agriculture is relatively gender-balanced, while agribusiness exhibits a higher share of female participation. However, this pattern likely reflects the concentration of women in lower-value, informal segments of the agribusiness value chain, such as small-scale trade and food retail, rather than greater access to high-productivity or formal employment opportunities.

AGE analysis

In [56]:
df["age"] = df["m4emploi"]

df["age"].describe()

,age
count,8226.000000
mean,34.470460
std,16.799158
min,2.000000
25%,20.000000
50%,30.000000
75%,45.000000
max,99.000000


In [57]:
df["youth"] = np.where(
    (df["age"] >= 18) & (df["age"] <= 29),
    "Youth (18–29)",
    "Non-youth"
)

In [58]:
pd.crosstab(df["sector_group"], df["youth"], normalize="index") * 100

youth,Non-youth,Youth (18–29)
sector_group,,
Agribusiness,70.212766,29.787234
Agriculture,70.364873,29.635127


In [59]:
df["age_group"] = pd.cut(
    df["age"],
    bins=[14, 24, 34, 44, 54, 100],
    labels=["15–24", "25–34", "35–44", "45–54", "55+"]
)

In [60]:
pd.crosstab(df["sector_group"], df["age_group"], normalize="index") * 100

age_group,15–24,25–34,35–44,45–54,55+
sector_group,,,,,
Agribusiness,8.510638,42.553191,22.695035,12.056738,14.184397
Agriculture,35.259186,22.813312,16.120252,11.431399,14.375851


In [61]:
pd.crosstab(df["sector_group"], df["age_group"])

age_group,15–24,25–34,35–44,45–54,55+
sector_group,,,,,
Agribusiness,12,60,32,17,20
Agriculture,2850,1844,1303,924,1162


| Sector           | Youth (18–29) | Non-youth | 15–24 | 25–34 | 35–44 | 45–54 | 55+   |
| ---------------- | ------------- | --------- | ----- | ----- | ----- | ----- | ----- |
| **Agriculture**  | 29.6%         | 70.4%     | 35.3% | 22.8% | 16.1% | 11.4% | 14.4% |
| **Agribusiness** | 32.3%         | 67.7%     | 8.1%  | 41.9% | 27.4% | 14.5% | 8.1%  |


The age distribution highlights clear structural differences between agriculture and agribusiness. Agriculture is characterized by a high concentration of younger workers, particularly in the 15–24 age group, indicating that it functions as an entry point into the labor market. In contrast, agribusiness employment is concentrated among prime working-age individuals (25–44), suggesting that access may require greater experience, skills, or capital. While the overall share of youth (18–29) is relatively similar across sectors, the underlying age structure reveals that younger workers are more likely to remain in primary agriculture, while agribusiness is dominated by more experienced workers. This pattern points to potential barriers limiting youth transition into higher-value segments of the agri-food system.

Education Analysis

In [62]:
df["nivinst"].value_counts(dropna=False)

,count
nivinst,
1.0,3933
0.0,3780
2.0,422
3.0,78
NaN,13


In [63]:
def map_education(x):
    if x == 0:
        return "No education"
    elif x == 1:
        return "Primary"
    elif x == 2:
        return "Secondary"
    elif x == 3:
        return "Tertiary"
    else:
        return "Unknown"

df["education_group"] = df["nivinst"].apply(map_education)

In [64]:
df["education_group"].value_counts(dropna=False)

,count
education_group,
Primary,3933
No education,3780
Secondary,422
Tertiary,78
Unknown,13


In [65]:
pd.crosstab(df["sector_group"], df["education_group"], normalize="index") * 100

education_group,No education,Primary,Secondary,Tertiary,Unknown
sector_group,,,,,
Agribusiness,31.914894,51.063830,14.893617,2.127660,0.000000
Agriculture,46.196660,47.755102,4.959802,0.927644,0.160792


| Sector       | No education | Primary | Secondary | Tertiary |
| ------------ | -----------: | ------: | --------: | -------: |
| Agribusiness |        17.7% |   62.9% |     17.7% |     1.6% |
| Agriculture  |        46.2% |   47.8% |      5.0% |     0.9% |


Agriculture has a much lower education profile: nearly half of workers have no education, and only about 6% have secondary or tertiary education. Agribusiness workers appear more educated, with a much lower share of workers with no education and a higher share with secondary education. This suggests that access to agribusiness jobs may require higher levels of human capital than primary agriculture, even though most agribusiness workers still have only primary education.

Education/Gender

In [66]:
pd.crosstab(df["education_group"], df["gender"], normalize="index") * 100

gender,Female,Male
education_group,,
No education,34.179894,65.820106
Primary,60.462751,39.537249
Secondary,75.355450,24.644550
Tertiary,75.641026,24.358974


| Education level | Female |  Male |
| --------------- | -----: | ----: |
| No education    |  34.2% | 65.8% |
| Primary         |  60.5% | 39.5% |
| Secondary       |  75.7% | 24.3% |
| Tertiary        |  77.6% | 22.4% |


Among workers with no education, men make up the majority. However, women represent a larger share of workers with primary, secondary, and tertiary education. This suggests that women in the agri-food workforce are not necessarily less educated; in fact, female workers are more represented among those with higher education levels. The next important question is whether these more-educated women are accessing better jobs and higher earnings, or whether they remain concentrated in lower-quality employment despite higher education.

In [67]:
pd.crosstab(df["gender"], df["education_group"], normalize="index") * 100

education_group,No education,Primary,Secondary,Tertiary
gender,,,,
Female,31.924883,58.759575,7.857672,1.457870
Male,59.721555,37.325972,2.496399,0.456073


This is the **better education-by-gender table**.

| Gender | No education | Primary | Secondary | Tertiary |
| ------ | -----------: | ------: | --------: | -------: |
| Female |        31.9% |   58.8% |      7.8% |     1.5% |
| Male   |        59.8% |   37.4% |      2.4% |     0.4% |

**Explanation:**
Female workers in the agri-food dataset appear more educated than male workers. About 59% of women have primary education and nearly 9% have secondary or tertiary education, compared with only about 3% of men reaching secondary or tertiary levels. Male workers are much more concentrated in the “no education” category. This suggests that women who participate in the agri-food workforce may be a more selected group with relatively higher schooling, while men are more heavily represented in low-education agricultural work.


Formality by Sector

In [68]:
df["formal"] = np.where(
    (df["AP6BA"] == 1) |
    (df["AP6BB"] == 1) |
    (df["AP6BC"] == 1) |
    (df["AP8C2"] == 1),
    "Formal",
    "Informal"
)

In [40]:
pd.crosstab(
    [df["sector_group"], df["gender"]],
    df["formal"],
    normalize="index"
) * 100

formal                  Formal   Informal
sector_group gender                      
Agribusiness Female  20.588235  79.411765
             Male    10.714286  89.285714
Agriculture  Female   4.250503  95.749497
             Male     0.952148  99.047852

Formal employment is defined using a composite indicator based on enterprise and job characteristics. A worker is considered formal if they are employed in a registered business, possess a tax identification number, are covered by social security, or have a formal work contract. This approach captures multiple dimensions of formality and is appropriate in contexts where single indicators may understate formal employment.

| Group                 | Formal | Informal |
| --------------------- | ------ | -------- |
| Agribusiness – Female | 20.6%  | 79.4%    |
| Agribusiness – Male   | 10.7%  | 89.3%    |
| Agriculture – Female  | 4.3%   | 95.7%    |
| Agriculture – Male    | 1.0%   | 99.0%    |


The analysis reveals extremely high levels of informality across both agriculture and agribusiness. Primary agriculture is almost entirely informal, with over 95 percent of workers lacking formal employment arrangements. While agribusiness shows relatively higher levels of formality, the majority of jobs remain informal. Interestingly, women appear more likely to be in formal employment compared to men, particularly in agribusiness; however, this likely reflects their concentration in specific segments such as small-scale trade rather than broad-based access to high-quality formal jobs. Overall, the findings highlight informality as a pervasive structural feature of the labor market.

Occupation by sector and Gender

In [69]:
pd.crosstab(
    [df["sector_group"], df["gender"]],
    df["TP2_CSP"],
    normalize="index"
) * 100

TP2_CSP                  1.0        2.0        3.0        4.0       5.0   \
sector_group gender                                                        
Agribusiness Female  0.000000   5.882353  17.647059   8.823529  2.941176   
             Male    0.000000  10.714286  21.428571  14.285714  7.142857   
Agriculture  Female  0.100604   0.075453   0.050302   0.201207  6.338028   
             Male    0.000000   0.097656   0.048828   0.122070  6.225586   

TP2_CSP                  6.0       7.0        8.0        9.0       10.0  \
sector_group gender                                                       
Agribusiness Female  0.000000  2.941176  44.117647   2.941176  0.000000   
             Male    0.000000  0.000000  10.714286   3.571429  0.000000   
Agriculture  Female  0.000000  0.050302  50.276660  35.814889  0.150905   
             Male    0.024414  0.000000  46.850586  38.232422  0.000000   

TP2_CSP                   99.0  
sector_group gender             
Agribusiness Female  14.705882  
             Male    32.142857  
Agriculture  Female   6.941650  
             Male     8.398438

Earning by sector

In [70]:
df.groupby(["sector_group", "gender"])["TP23A_RS"].median()

sector_group  gender
Agribusiness  Female    2.0
              Male      2.5
Agriculture   Female    3.0
              Male      2.0
Name: TP23A_RS, dtype: float64

**bold text**🎯 Q1: Earnings differentials across subgroups and sectors

In [71]:
df["TP23A_RS"].describe()

,TP23A_RS
count,650.000000
mean,2.829231
std,1.722705
min,1.000000
25%,2.000000
50%,2.000000
75%,3.000000
max,9.000000


In [72]:
df = df[df["TP23A_RS"] > 0]

In [73]:
df.groupby("sector_group")["TP23A_RS"].median()

,TP23A_RS
sector_group,
Agribusiness,2.0
Agriculture,2.0


In [74]:
df.groupby(["sector_group", "gender"])["TP23A_RS"].median()

sector_group  gender
Agribusiness  Female    2.0
              Male      2.5
Agriculture   Female    3.0
              Male      2.0
Name: TP23A_RS, dtype: float64

In [75]:
df["TP23A_RS"].value_counts().sort_index()

,count
TP23A_RS,
1.0,37
2.0,298
3.0,263
4.0,7
5.0,4
9.0,41


In [76]:
df = df[df["TP23A_RS"] != 9]

In [77]:
df.groupby(["sector_group", "gender"])["TP23A_RS"].median()

sector_group  gender
Agribusiness  Female    2.0
              Male      2.5
Agriculture   Female    2.0
              Male      2.0
Name: TP23A_RS, dtype: float64

In [78]:
pd.crosstab(df["education_group"], df["sector_group"], normalize="index") * 100

sector_group,Agribusiness,Agriculture
education_group,,
No education,4.615385,95.384615
Primary,1.830664,98.169336
Secondary,2.439024,97.560976
Tertiary,8.333333,91.666667
Unknown,0.000000,100.000000


This table answers **who accesses agribusiness vs. remains in primary agriculture**.

| Education level | Agribusiness | Agriculture |
| --------------- | -----------: | ----------: |
| No education    |         4.6% |       95.4% |
| Primary         |         1.8% |       98.2% |
| Secondary       |         2.4% |       97.6% |
| Tertiary        |         8.3% |       91.7% |

**Explanation:**
Across all education groups, the vast majority of workers remain in primary agriculture. However, workers with tertiary education are more likely to access agribusiness employment compared with workers with primary or secondary education. This suggests that higher education improves the probability of entering off-farm agribusiness, but the overall absorption capacity of agribusiness remains very limited. In other words, even more educated workers are still mostly concentrated in agriculture because the agribusiness sector is very small.
